app/controllers/snippets_controller.rb

```diff
@@ -50,6 +51,15 @@ def update
 
   private
 
+  def post_slack_message
+    client = Slack::Web::Client.new
+    puzzle = Puzzle.by_date(Aoc.begin_time.change(day: params[:day]))
+    username = "<#{helpers.profile_url(current_user.uid)}|#{current_user.username}>"
+    solution = "<#{helpers.snippet_url(day: @snippet.day, challenge: @snippet.challenge, anchor: @snippet.id)}|solution>"
+    text = "#{username} submitted a new #{solution} for part #{params[:challenge]} in :#{@snippet.language}-hd:"
+    client.chat_postMessage(channel: ENV.fetch("SLACK_CHANNEL", "#aoc-dev"), text:, thread_ts: puzzle.thread_ts)
+  end
+
```

Comment:
We should see to extract this to a `Snippets::NotifyToSlack` class instead at some point. That would also remove the need to have `@snippet` be an instance variable in the controller.

Reply:
Captured in #682 👍 thanks!

Reply:
Ca peu etre fait dans https://github.com/pil0u/lewagon-aoc/pull/678